src/commands/staff/mute.ts

```diff
@@ -10,23 +9,40 @@ export default class extends Command {
 	usage = '<user>';
 	runInDM = false;
 
-	permissions(msg: Message): boolean {
-		return staffPerms(msg);
-	}
+	tempPermissions: ApplicationCommandPermissionData[] = [STAFF_PERMS, ADMIN_PERMS];
+
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'user',
+			description: 'The user to mute',
+			type: 'USER',
+			required: true
+		}
+	]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const user = interaction.options.getUser('user');
+		const member = await interaction.guild.members.fetch(user.id);
+
+		if (!member) {
+			interaction.reply({ content: `Something went wrong. Please contact ${MAINTAINERS} for help.`, ephemeral: true });
+			throw new Error('Could not find member based on passed in user');
+		}
 
-	async run(msg: Message, [member]: [GuildMember]): Promise<Message> {
 		if (member.roles.cache.has(ROLES.MUTED)) {
-			const reason = `${member.user.username} was un-muted by ${msg.author.tag} (${msg.author.id})`;
-			member.roles.remove(ROLES.MUTED, reason);
-			return msg.channel.send(`${member.user.username} has been un-muted.`);
+			const reason = `${member.user.username} was un-muted by ${interaction.user.tag} (${interaction.user.id})`;
+			await member.roles.remove(ROLES.MUTED, reason);
+			return interaction.reply({ content: `${member.user.username} has been un-muted.`, ephemeral: true });
 		}
-		const reason = `${member.user.username} was muted by ${msg.author.tag} (${msg.author.id})`;
-		member.roles.add(ROLES.MUTED, reason);
-		return msg.channel.send(`${member.user.username} has been muted.`);
+		const reason = `${member.user.username} was muted by ${interaction.user.tag} (${interaction.user.id})`;
+		await member.roles.add(ROLES.MUTED, reason);
+		return interaction.reply({ content: `${member.user.username} has been muted.`, ephemeral: true });
 	}
 
-	async argParser(msg: Message, input: string): Promise<Array<GuildMember>> {
-		return [await userParser(msg, input)];
+	permissions(msg: Message): boolean {
+		return staffPerms(msg);
 	}
 
+	async run(_msg: Message): Promise<Message> { return; }
+
 }
```

Comment:
For this command I'd also recommend possibly DMing the mute victim with their mute reason once the mute goes through. (Since you cannot send ephemeral, non-reply messages)

Reply:
It emails anyway